CameraOverlay/overlay_all_stats.py

```diff
@@ -42,109 +25,69 @@ def on_connect(self, client, userdata, flags, rc):
-			self.data["reed_velocity"] += float(parsed_data["reed_velocity"])
-			self.data["count"] = self.data["count"] + 1
-			total_time = current_time - self.start_time
-			update_time = 0.5
-
-			if total_time >= update_time:
-				self.start_time = current_time
-				self.data_canvas.clear()
-
-				# Display power
-				if self.data["power"] != 0:
-					power = self.data["power"] / self.data["count"]
-					rec_power = self.data["rec_power"]
-					tolerance = 0.05
-					# Display recommended power
-					self.data_canvas.draw_text("{0}".format(round(rec_power, 2)), (340, self.text_height * 1))
-					# Display power
-					if power > (rec_power + (rec_power * tolerance)):
-						power_colour = Colour.red
-					elif power > rec_power:
-						power_colour = Colour.green
-					else:
-						power_colour = Colour.black
-					self.data_canvas.draw_text("{0}".format(round(power, 2)), (340, self.text_height * 2), colour=power_colour)
-
-				# Display cadence
-				if self.data["cadence"] != 0:
-					cadence = self.data["cadence"] / self.data["count"]
-					self.data_canvas.draw_text("{0}".format(round(cadence, 2)), (340, self.text_height * 3))
-
-				# Display speed
-				if self.data["reed_velocity"] != 0:
-					# Max Speed
-					max_speed = self.data["max_speed"]
-					max_speed_text = "{0} km/h".format(round(max_speed, 2))
-					max_speed_pos = (self.width // 2 - 70, self.height - self.speed_height * 2)
-					self.data_canvas.draw_text(max_speed_text, max_speed_pos, size=2.5)
-
-					# Recommended speed
-					rec_speed = self.data["rec_speed"]
-					rec_speed_text = "{0} km/h".format(round(rec_speed, 2))
-					rec_speed_pos = (self.width // 2 - 70, self.height - self.speed_height * 1)
-					self.data_canvas.draw_text(rec_speed_text, rec_speed_pos, size=2.5)
-
-					# Actual speed
-					speed = self.data["reed_velocity"] / self.data["count"]
-					speed_text = "{0} km/h".format(round(speed, 2))
-					speed_pos = (self.width // 2 - 70, self.height - self.speed_height * 0)
-					tolerance = 0.05
-					if speed > (rec_speed + (rec_speed * tolerance)):
-						speed_colour = Colour.red
-					elif speed > rec_speed:
-						speed_colour = Colour.green
-					else:
-						speed_colour = Colour.black
-					self.data_canvas.draw_text(speed_text, speed_pos, colour=speed_colour, size=2.5)
-
-				# Display reed_distance (distance travelled)
-				if self.data["reed_distance"] != 0:
-					reed_distance = self.data["reed_distance"] / self.data["count"]
-					self.data_canvas.draw_text("{0}".format(round(reed_distance, 2)), (340, self.text_height * 4))
-
-				# Reset variables
-				self.data["power"] = 0
-				self.data["cadence"] = 0
-				self.data["gps_speed"] = 0
-				self.data["reed_velocity"] = 0
-				self.data["reed_distance"] = 0
-				self.data["count"] = 0
+	def update_data_layer(self):
+		self.data_canvas.clear()
+		self.message_canvas.clear()
+
+		# Display power
+		if self.data["power"] != 0:
+			power = self.data["power"]
```

Comment:
Should we create function handlers for each data type? Will help with cleaning up this method and making it easier to read.